src/typing/env_js.ml

```diff
@@ -327,7 +401,7 @@ let string_of_env cx ctx =
 let install_refinement cx x xtypes =
   if exists_in_scope x (peek_env ()) = None then
     let t = SMap.find_unsafe x xtypes in
-    init_env cx x (create_env_entry t t None)
+    init_env cx x (create_env_entry t t None) VarScope (* TODO: verify correct hoist behavior *)
```

Comment:
I've implemented this in 4659a41ed64ba840609a5c0a032612a2a90224cd. Mostly sure it's correct :)